### DISCUSSION TITLE
dynamic route in next js

### DISCUSSION THREAD

**Discussion by farouk46015** [vercel/next.js#12260] [Help] [answered]
hi 
i get this error in console 
net::ERR_ABORTED 404 (Not Found)
for all dynamic  pages  i am trying to assess by Link 
 i put link in this way 
`<Link href={`/${props.model}/${props.slug}`} as={`/${props.model}/${props.slug}`} > `

and i have  file [slug].js 
i works but i get the error in console because browser request file as static files  

i am using next 9.1 
<img width="1680" alt="Screen Shot 2020-04-28 at 00 33 39" src="https://user-images.githubusercontent.com/32044283/80423112-0d96d380-88e8-11ea-8895-b12eb375d4a7.png">

i need to know best way of writing slug to get dynamic pages

**Comment by ties-v** [comment, marked as answer]
You need to set the `href` prop to the "real file" and use the `as` prop to fill in the variables, see https://nextjs.org/docs/api-reference/next/link#dynamic-routes

So in your case something like
```js
<Link href="/[model]/[slug]" as={`/${props.model}/${props.slug}`}>
```
should work as it now routes to `[model]/[slug].js` file.

**Reply by farouk46015** [reply]
i have tried this way i still get the same error it works in both but with error message

```
  <Link href={`/${props.model}/[slug]`} as={`/${props.model}/${props.slug}`}> 
      <a className="d-block">
           <h1>{props.title}</h1>
       </a>    
  </Link>
```

**Reply by timneutkens** [reply]
```
<Link href={`/[model]/[slug]`} as={`/${props.model}/${props.slug}`}> 
      <a className="d-block">
           <h1>{props.title}</h1>
       </a>    
  </Link>
```

**Reply by farouk46015** [reply]
than you 
i works
